ui/components/src/UnitSelector/UnitSelector.test.tsx

```diff
@@ -0,0 +1,193 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { UnitOptions } from '../model';
+import { UnitSelector } from './UnitSelector';
+
+describe('UnitSelector', () => {
```

Comment:
Yeah what you mentioned in the description are all bugs that aren’t being tracked anywhere, I thought the decimal_places rounding was working at one point, I should have added some better tests like what you’ve done in this PR!

Reply:
I added a quick fix in PR #747 that partially fixes the decimal_places bug, going to come back to it later though